### PR TITLE
Add documentation about decorators

### PR DESCRIPTION
**Pull request by laurencap**
Add decorator documentation, link from README

### CODE REVIEW

File: src/braket/experimental/autoqasm/doc/decorators.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+bell(2, 3);
+```
+
+## `@aq.gate`
+
+Represents a gate definition.
+
+Gate definitions define higher-level gates with support gates, and are often used to decompose a gate into the native gates of a device.
+
+The body of a gate definition can only contain gates. Qubits used in the body of a gate definition must be passed as input arguments, with the type hint `aq.Qubit`. Like subroutines, a gate must be called by a main quantum program to have an effect.
```
Comment (rmshaffer) [inline src/braket/experimental/autoqasm/doc/decorators.md:76]:
```suggestion
The body of a gate definition can only contain gates. Qubits used in the body of a gate definition must be passed as input arguments, with the type hint `aq.Qubit`. Like subroutines, a gate definition must be called from within the context of a main quantum program or subroutine in order to be included in the program.
```

File: src/braket/experimental/autoqasm/doc/decorators.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+```
+@aq.gate
+def ch(q0: aq.Qubit, q1: aq.Qubit):
+    """Define a controlled-Hadamard gate."""
+    ry(q1, -math.pi / 4)
+    cz(q0, q1)
+    ry(q1, math.pi / 4)
```
Comment (rmshaffer) [inline src/braket/experimental/autoqasm/doc/decorators.md:84]: I like this example!

File: src/braket/experimental/autoqasm/doc/decorators.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    pulse.shift_frequency(q0_rf_frame, -321047.14178613486)
+    pulse.play(q0_rf_frame, waveform(angle))
+    pulse.shift_frequency(q0_rf_frame, 321047.14178613486)
+    pulse.barrier("$0")
+    
+@aq.main
+def my_program():
+    rx(0, 0.123)
+    measure(0)
+```
```
Comment (rmshaffer) [inline src/braket/experimental/autoqasm/doc/decorators.md:123]: Could you also include the example of calling `with_calibrations()` to attach the calibration?

File: src/braket/experimental/autoqasm/doc/decorators.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    pulse.barrier("$0")
+    pulse.shift_frequency(q0_rf_frame, -321047.14178613486)
+    pulse.play(q0_rf_frame, waveform(angle))
+    pulse.shift_frequency(q0_rf_frame, 321047.14178613486)
+    pulse.barrier("$0")
+    
+@aq.main
+def my_program():
+    rx(0, 0.123)
+    measure(0)
```
Comment (rmshaffer) [inline src/braket/experimental/autoqasm/doc/decorators.md:122]: If the intent is to match the calibration (?), the program should use `"$0"`:
```suggestion
    rx("$0", 0.123)
    measure("$0")
```

File: src/braket/experimental/autoqasm/doc/decorators.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+
+## `@aq.gate_calibration`
+
+This decorator allows you to register a calibration for a gate. A gate calibration is a device-specific, low-level, pulse implementation for a logical gate operation.
+
+At the pulse level, qubits are no longer interchangable. Each one has unique properties. Thus, a gate calibration is usually defined for a concrete set of qubits and parameters, but you can use input arguments to your function as well.
+
+The body of a gate calibration must only contain pulse operations. This decorator requires one input arguments to specify the `Gate` that the calibration will be registered to. Concrete values for the qubits and parameters are supplied as keyword arguments to the decorator.
+The union of the arguments of the decorator and the decorated function must match the arguments of the gate to be implemented.
```
Comment (rmshaffer) [inline src/braket/experimental/autoqasm/doc/decorators.md:102]:
```suggestion
Every qubit and angle parameter of the gate being implemented must appear either as an argument to the `@aq.gate_calibration` decorator, or as a parameter of the decorated function.
```

File: src/braket/experimental/autoqasm/doc/decorators.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+```
+
+
+## `@aq.gate_calibration`
+
+This decorator allows you to register a calibration for a gate. A gate calibration is a device-specific, low-level, pulse implementation for a logical gate operation.
+
+At the pulse level, qubits are no longer interchangable. Each one has unique properties. Thus, a gate calibration is usually defined for a concrete set of qubits and parameters, but you can use input arguments to your function as well.
+
+The body of a gate calibration must only contain pulse operations. This decorator requires one input arguments to specify the `Gate` that the calibration will be registered to. Concrete values for the qubits and parameters are supplied as keyword arguments to the decorator.
```
Comment (rmshaffer) [inline src/braket/experimental/autoqasm/doc/decorators.md:101]:
```suggestion
The body of a function decorated with `@aq.gate_calibration` must only contain pulse operations.

The first argument to the `@aq.gate_calibration` decorator must be the gate function that the calibration will be registered to. Concrete values for the qubits and parameters are supplied as keyword arguments to the decorator.
```

**Review by rmshaffer — APPROVED** [review]
Looks great, thanks for adding this! I left a few suggestions.

File: src/braket/experimental/autoqasm/doc/decorators.md (hunk 2 above)
Comment (laurencap) [reply src/braket/experimental/autoqasm/doc/decorators.md:84]: Can't take credit! @yitchen-tim

**Pull request merged**